docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -0,0 +1,341 @@
+
+A software-defined asset includes three main components:
+
+- An <PyObject object="AssetKey" />, which is a handle for referring to the asset.
+- A set of upstream asset keys, which refer to assets that the contents of the software-defined asset are derived from.
+- An [op](/concepts/ops-jobs-graphs/ops), which is a function responsible for computing the contents of the asset from its upstream dependencies.
+
+A crucial distinction between software-defined assets and [ops](/concepts/ops-jobs-graphs/ops) is that software-defined assets know about their dependencies, while ops do not. Ops aren't hooked up to dependencies until they're placed inside a [graph](/concepts/ops-jobs-graphs/jobs-graphs).
+
+"Materializing" an asset is the act of running its op and saving the results to persistent storage. You can initiate materializations from [Dagit](/concepts/dagit/dagit), Dagster's web UI, or by invoking Python APIs. By default, assets are materialized to pickle files on your local filesystem, but materialization behavior is [fully customizable](#customizing-how-assets-are-materialized-with-io-managers).
+
+A single software-defined asset might be represented in multiple storage environments - e.g. it might have "production" version and a "staging" version.
+
+## Defining assets
+
+### A basic software-defined asset
+
+The easiest way to create a software-defined asset is with the <PyObject object="asset" decorator /> decorator.
+
+```python file=/concepts/assets/basic_asset_definition.py
+from dagster import asset
+
+
+@asset
+def my_asset():
+    return [1, 2, 3]
+```
+
+By default, the name of the decorated function, `my_asset`, is used as the asset key. The decorated function forms the asset's op: it's responsible for producing the asset's contents. This asset doesn't depend on any other assets.
+
+### Assets with dependencies
+
+Software-defined assets can depend on other software-defined assets. The easiest way to define an asset dependency is to include an upstream asset name as an argument to the decorated function. In the following example, `downstream_asset` depends on `upstream_asset`. That means that the contents of `upstream_asset` are provided to the function that computes the contents of `downstream_asset`.
+
+```python file=/concepts/assets/asset_dependency.py startafter=start_marker endbefore=end_marker
+@asset
+def upstream_asset():
+    return [1, 2, 3]
+
+
+@asset
+def downstream_asset(upstream_asset):
+    return upstream_asset + [4]
+```
+
+The [explicit dependencies](#explicit-dependencies) example covers an alternative way to specify asset dependencies without needing to match argument names to upstream asset names.
+
+## Asset groups
```

Comment:
maybe "Grouping assets" or "Linking assets". "Asset groups" isn't clear to me what i can accomplish by using it.

I'm assuming "Asset Group" was a term but we changed it to asset collection, or the other way around? Either way, using some verb as the heading could be more clear and also consistent with other subsections

Reply:
The other way around - I'll add a verb